katsdpimager/test/test_grid.py

```diff
@@ -148,20 +147,23 @@ def callback(max_vis, vis):
                 self.array_parameters,
                 self.image_parameters,
                 self.grid_parameters,
-                max_vis, allocator=accel.SVMAllocator(context))
+                max_vis)
             fn.ensure_all_bound()
             grid_data = fn.buffer('grid')
-            grid_data.fill(0)
+            grid_data.zero(command_queue)
             weights_grid = fn.buffer('weights_grid')
             weights_grid_host = weights_grid.empty_like()
             _middle(weights_grid_host, self.weights_grid.shape)[:] = self.weights_grid
             weights_grid.set_async(command_queue, weights_grid_host)
-            fn.num_vis = len(self.uv)
-            fn.set_coordinates(self.uv, self.sub_uv, self.w_plane)
-            fn.set_vis(vis)
+            n_vis = len(self.uv)
+            fn.num_vis = n_vis
+            fn.buffer('uv').set_region(
```

Comment:
This is a bit more obtuse than before, but I guess more explicit. Would there be a benefit to keep the original methods and just use `.set_region()` inside?

Reply:
I wanted to have some separation of concerns such that all transfers routed through the top-level Imaging class, since often a buffer is shared between several lower-level classes and hence no one of them "owns" the buffer.